### PR TITLE
chore: SSO refactor

### PR DESCRIPTION
**Pull request by jfaracik**
The purpose of this PR is to remove the amount of unused files, consolidate authentication into a single app, and make code more reusable.

### CODE REVIEW

File: conf/middleware.py [hunk 1]
```diff
@@ -15,8 +17,8 @@ def __init__(self, get_response):
         self.get_response = get_response
 
     def __call__(self, request):
-        if resolve(request.path).app_name != "authbroker_client" and not request.user.is_authenticated:
-            return redirect("authbroker_client:login")
+        if resolve(request.path).app_name != "auth" and not request.user.is_authenticated:
```
Comment (BenAndrew-Kainos) [inline conf/middleware.py]: Could you store "auth" as a constant

File: ui_automation_tests/step_defs/test_users_steps.py [hunk 2]
```diff
@@ -23,8 +20,6 @@ def add_user(driver):
         for x in range(3):
```
Comment (BenAndrew-Kainos) [inline ui_automation_tests/step_defs/test_users_steps.py]: Who did this and why do we need it?

File: users/forms.py [hunk 3]
```diff
@@ -24,20 +21,18 @@ def add_user_form(request):
     )
 
 
-def edit_user_form(request, user_id, super_user: bool):
+def edit_user_form(request, user):
     return Form(
-        title=strings.USER_EDIT_TITLE,
+        title="Change user role",
```
Comment (BenAndrew-Kainos) [inline users/forms.py]: Move to strings

File: users/services.py [hunk 4]
```diff
@@ -26,4 +26,7 @@ def update_user(request, pk, json):
 
 
 def is_super_user(user):
```
Comment (BenAndrew-Kainos) [inline users/services.py]: Could you change function usage instead to pass `user["user"]` where applicable

File: ui_automation_tests/step_defs/test_users_steps.py (hunk 2 above)
Comment (jfaracik) [reply ui_automation_tests/step_defs/test_users_steps.py]: I'm not too sure.

**Pull request merged**